Cubical/Basics/CartesianKanOps.agda

```diff
@@ -56,43 +43,77 @@ coei0→0 A a = refl
+-- unlike in cartesian cubes, we don't get coei→i = id definitionally
 coei→j : ∀ {ℓ} (A : I → Set ℓ) (i j : I) → A i → A j
-coei→j A i j a = transp (λ k → A ((i ∧ ~ k) ∨ (j ∧ k))) (~ i ∧ ~ j) a
+coei→j A i j a =
+  fill A
+    (λ j → λ { (i = i0) → coe0→i A j a
+             ; (i = i1) → coe1→i A j a
+             })
+    (inc (coei→0 A i a))
+    j
+
+-- "squeeze"
+-- this is just defined as the composite face of the master coe
+coei→1 : ∀ {ℓ} (A : I → Set ℓ) (i : I) → A i → A i1
+coei→1 A i a = coei→j A i i1 a
+
+coei0→1 : ∀ {ℓ} (A : I → Set ℓ) (a : A i0) → coei→1 A i0 a ≡ coe0→1 A a
+coei0→1 A a = refl
 
--- We only get the two equations for i0:
+coei1→1 : ∀ {ℓ} (A : I → Set ℓ) (a : A i1) → coei→1 A i1 a ≡ a
+coei1→1 A a = refl
 
+-- equations for "master coe"
 coei→i0 : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i) → coei→j A i i0 a ≡ coei→0 A i a
 coei→i0 A i a = refl
 
 coei0→i : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i0) → coei→j A i0 i a ≡ coe0→i A i a
 coei0→i A i a = refl
 
--- For i1 we don't get the equations definitionally:
-
--- coei→i1 : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i) → coei→j A i i1 a ≡ coei→1 A i a
--- coei→i1 A i a = {!!}
-
--- coei1→i : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i1) → coei→j A i1 i a ≡ coe1→i A i a
--- coei1→i A i a = {!!}
-
-
--- We can also get the equations for i1, but then not for i0
-
-coei→j' : ∀ {ℓ} (A : I → Set ℓ) (i j : I) → A i → A j
-coei→j' A i j a = transp (λ k → A ((i ∧ j) ∨ (j ∧ k) ∨ (~ k ∧ i))) (i ∧ j) a 
-
-coei→i1' : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i) → coei→j' A i i1 a ≡ coei→1 A i a
-coei→i1' A i a = refl
-
-coei1→i' : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i1) → coei→j' A i1 i a ≡ coe1→i A i a
-coei1→i' A i a = refl
-
--- coei→i0' : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i) → coei→j' A i i0 a ≡ coei→0 A i a
--- coei→i0' A i a = {!!}
-
--- coei0→i' : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i0) → coei→j' A i0 i a ≡ coe0→i A i a
--- coei0→i' A i a = {!!}
+coei→i1 : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i) → coei→j A i i1 a ≡ coei→1 A i a
+coei→i1 A i a = refl
+
+coei1→i : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i1) → coei→j A i1 i a ≡ coe1→i A i a
+coei1→i A i a = refl
+
+-- only non-definitional equation
+coei→i : ∀ {ℓ} (A : I → Set ℓ) (i : I) (a : A i) → coei→j A i i a ≡ a
+coei→i A i = coe0→i (λ i → (a : A i) → coei→j A i i a ≡ a) i (λ _ → refl)
+
+-- We can reconstruct fill from hfill, coei→j, and the path coei→i ≡ id.
+-- The definition does not rely on the computational content of the coei→i path.
+
+-- * Note: the definition of fill' below DOES use fill, but only under the constraint φ ∨ ~i, where it will
+-- reduce either to the cap or the tube. We only have to do this because of technical limitations of cubical
+-- Agda; where we write
+--
+--   λ j _ → coei→i A i (fill A u u0 i) j
+--
+-- what we really want to write is
+--
+--   "λ j → λ {φ → coei→i A i (u i); (i = i0) → coei→i A i (ouc u0)}"
+--
+-- but this is (apparently?) not possible.
```

Comment:
@Saizan Can you comment on this?

Reply:
it is completely possible, I was the fool...